modules/discord/Classes/Discord.php

```diff
@@ -46,25 +46,18 @@ public function __construct($discordServerId = ''){
      */
     
     public function fetchServerData(){
-        global $cfg;
+        global $cfg, $cache;
```

Comment:
Can we use a proper Dependency injection here?
global variables make it harder and harder. I know that we have them all over the place right now, but while introducing a new system we can make it better.

Reply:
I do see that dependency injection has its uses and also that global variables should be avoided unless there is a good reason for them.
But: I would argue that this is case where they make sense.
This is a static, non-changing object reference that is used everywhere, so I do see the benefit of one global instance.

> global variables make it harder and harder.

What exactly?
I would hate to see functions where the first five or six parameters are only global dependencies.
Also: Refactoring is quite simple with globals. Just search for the variable name....

In case of an individual set of data/object, it should be passed as argument, complete agreement there.

Reply:
If Dependency Injection was used in combination with the [PSR Simple Cache Interface (](https://packagist.org/packages/psr/simple-cache) Modules could decide on their own if they prefer f.e. file, database or in memory cacheing.

Reply:
> If Dependency Injection was used in combination with the [PSR Simple Cache Interface ](https://packagist.org/packages/psr/simple-cache) Modules could decide on their own if they prefer f.e. file, database or in memory caching.

As the cache implementation is done in each module separately that is possible in any case.
Also with DI it wouldn't be the module that decides what cache is used, but the caller...

If a module needs a separate cache pool implementation, then that is possible as the required sources are included with this PR forward.
But if just a simple cache is required, then this code provides that.